postal/service.go

```diff
@@ -220,6 +228,25 @@ func (s Service) Resolve(path, id, version, stack string) (Dependency, error) {
 	return compatibleVersions[0], nil
 }
 
+func isCorrectArch(dep Dependency) bool {
+	systemArch := archFromSystem()
+
+	if systemArch == "amd64" && dep.Arch == "" {
+		return true
+	}
+
+	return systemArch == dep.Arch
+}
+
+func archFromSystem() string {
```

Comment:
In terms of clean code, one function should only do one thing, I would remove from this function, the two things that is doing 1. looking if the environment variable exists 2. fetching the system architecture. Changing that function to fetching only the architecture does not make sense as there is already a function for that. Therefore, I would suggest to move this logic out in to the Parent function (`Resolve`) as I think this is totally normal as there is already logic for validating other things like (`dependency version` and `stacks`). Althouh, if you decide to fetch the platform, then it would be good to wrap these two lines in a function called `getSystemPlatform` as it will be more than one line (one for fetching the `GOARCH` and one for fetching the `GOOS`

Reply:
I think we dont want to default into the system that the user is using as pack currently fallbacks to linux/amd64, so I would choose this platform as default, in order to avoid any errors from macos users that currently building applications

Reply:
> I think we dont want to default into the system

I might be completely wrong, but that is not what is done. It looks up the architecture of the runtime, meaning the buildpack that actually called `getDependency()`. So no other architecture could even work. The `BP_ARCH` is only needed for unit tests.

I tried to keep everything as similar to [`libak`](https://github.com/paketo-buildpacks/libpak/blob/aa7107c9e10714dce0adc2a13d85aa26cd739253/buildpack.go#L566) as possible.